saleor/graphql/discount/mutations/promotion/promotion_rule_create.py

```diff
@@ -64,6 +64,8 @@ def clean_input(
         cls, info: ResolveInfo, instance: models.PromotionRule, data: dict, **kwargs
     ):
         cleaned_input = super().clean_input(info, instance, data, **kwargs)
+        if "gift_ids" in cleaned_input:
+            cleaned_input["gifts"] = cleaned_input.pop("gift_ids")
```

Comment:
It looks like [validation](https://github.com/saleor/saleor/pull/15274/files#diff-0c7d12fd6312a1ac0904514edd8991f068e1ff699d238fd49a7adcd83573e87eR278) requires the instances, but here we will have just ids

Reply:
Actually `ModelMutation.clean_input` does the trick.